### PR TITLE
Proposal for implementation of issue #22628(improvements in sim::OnePhoton)

### PR DESCRIPTION
**Pull request by PetrilloAtWork**
Proposal for implementation of [issue #22628](https://cdcvs.fnal.gov/redmine/issues/22628) (improvements in `sim::OnePhoton`).

### CODE REVIEW

**Comment by aihimmel** [comment]
The principle seems sound, and if I understand it correctly this should not change results, just prevent us from making mistakes with world vs local vectors in the future. Probably fine, but let’s wait for the tests to come back.

File: lardataobj/Simulation/SimPhotons.h [hunk 1]
```diff
@@ -1,160 +1,274 @@
...
+// -----------------------------------------------------------------------------
+// forward declarations
+namespace sim {
+  class OnePhoton;
+  class SimPhotons;
+  class SimPhotonsLite;
+  class SimPhotonsCollection;
+
+  /// `a` is smaller than `b` if has earlier `Time`, or lower `MotherTrackID`.
+  bool operator< (OnePhoton const&, OnePhoton const&);
```
Comment (knoepfel) [inline lardataobj/Simulation/SimPhotons.h:54]: Recommend making this `noexcept`.

File: lardataobj/Simulation/SimPhotons.h [hunk 2]
```diff
@@ -1,160 +1,274 @@
...
+
+// -----------------------------------------------------------------------------
+// ---  sim::SimPhotons
+// -----------------------------------------------------------------------------
+
+inline int sim::SimPhotons::OpChannel() const { return fOpChannel; }
+
+inline void sim::SimPhotons::SetChannel(int ch) { fOpChannel = ch; }
+
+inline bool sim::SimPhotons::operator== (const sim::SimPhotons& other) const
```
Comment (knoepfel) [inline lardataobj/Simulation/SimPhotons.h:258]: Recommend making `noexcept`.

**Review by knoepfel — COMMENTED** [review]
Bravo, Gianluca.  Nice cleanup.  Only recommendation would be to make the functions `noexcept` where possible.

**Comment by knoepfel** [comment]
Approve

**Pull request merged**